src/Share/Names/Postgres.hs

```diff
@@ -0,0 +1,38 @@
+-- | Efficiently fetch a Names object for a given set of labeled dependencies.
+module Share.Names.Postgres (namesForReferences) where
+
+import Control.Lens
+import Data.Set qualified as Set
+import Share.Postgres qualified as PG
+import Share.Postgres.NameLookups.Conversions qualified as CV
+import Share.Postgres.NameLookups.Ops qualified as NameLookupOps
+import Share.Postgres.NameLookups.Types (NamesPerspective)
+import Share.Postgres.NameLookups.Types qualified as NameLookups
+import Share.Postgres.Refs.Types
+import Share.Prelude
+import Unison.LabeledDependency (LabeledDependency)
+import Unison.Name (Name)
+import Unison.Names (Names)
+import Unison.Names qualified as Names
+import Unison.Reference qualified as V1
+import Unison.Referent qualified as V1
+
+namesForReferences :: forall m. (PG.QueryM m) => NamesPerspective -> Set LabeledDependency -> m Names
+namesForReferences namesPerspective refs = do
+  withPGRefs <-
+    Set.toList refs
+      & CV.labeledDependencies1ToPG
+  (termNames, typeNames) <- foldMapM namesForReference withPGRefs
+  pure $ Names.fromTermsAndTypes termNames typeNames
+  where
+    -- TODO: Can probably speed this up by skipping suffixification.
```

Comment:
Probably worth testing one of the common queries on prod with and without suffixification to see if the real-world cost changes :)

Reply:
Yeah, how might we go about doing this? (Btw, this was one of the functions you authored before I took over)

Reply:
If you need a hand knowing where to poke around don't hesitate to ask, it's probably good to get more than me used to digging around :)